### DISCUSSION TITLE
Executing java script before snapshot

### DISCUSSION THREAD

**Discussion by s-petrov8** [percy/percy-selenium-dotnet#32] [Q&A] [answered]
Greetings!

Is it possible to pass some java script to execute on the page before taking a snapshot?
Similar to the 'execute' option for the snapshot command for the Percy CLI, described here: https://docs.percy.io/docs/percy-snapshot#per-snapshot

Tried like this, no result:

```
Percy.Options snapshotOptions = new();
snapshotOptions.Add("execute", "document.querySelector('[data-testid=\'promotion-container\']').remove();");
Percy.Snapshot((WebDriver)Driver.NativeDriver, "Basket with list options item", snapshotOptions);
```

I am trying to remove a dynamic element from the page that is causing snapshot comparisons to fail.

Kind regards,
Stoyan

**Comment by Robdel12** [comment, marked as answer]
Hey @s-petrov8! In SDKs that have full on test harnesses you would execute JS before calling `Percy.Snapshot`. In this specific example you would use Selenium's js execute before calling `Percy.Snapshot`

**Comment by s-petrov8** [comment]
Hi @Robdel12,

I assumed the 'execute' option would execute the java script only on the Percy side when processing the snapshots, thereby leaving the page in the Selenium tests unaffected. That would have been ideal for my case since I am interacting with the element later in my tests and it makes no sense to hide it.

Your suggestion is possible as a work around though! Thanks for your reply!
